src/net_processing.cpp

```diff
@@ -1723,6 +1730,12 @@ void PeerManagerImpl::BlockConnected(const std::shared_ptr<const CBlock>& pblock
             m_txrequest.ForgetTxHash(ptx->GetWitnessHash());
         }
     }
+    auto stalling_timeout = m_block_stalling_timeout.load();
+    // In case the dynamic timeout was doubled once or more, reduce it slowly back to its default value
+    const auto new_timeout = std::max(std::chrono::duration_cast<std::chrono::seconds> (stalling_timeout * 0.85), DEFAULT_BLOCK_STALLING_TIMEOUT);
```

Comment:
Probably should be IMO. Not sure how efficient std::atomic is with std::chrono units.

Reply:
> Is it intentional that this happens with an accuracy of 1 second?

It was kind of intentional, I first thought of changing `m_block_stalling_timeout` to microseconds, requiring conversions for the logging etc. but then I wondered whether a higher accuracy adds anything - I also thought of simply decreasing it constantly by one second per block received (instead of using a factor).
Do you or others have a preference here?

> The stalling_timeout != new_timeout condition can also be placed before the exchange, I think?

Done - I changed the order within the `if` statements (in two places).